tools/donate-cpu-server.py

```diff
@@ -142,37 +143,70 @@ def crashReport(results_path):
     html = '<html><head><title>Crash report</title></head><body>\n'
     html += '<h1>Crash report</h1>\n'
     html += '<pre>\n'
-    html += '<b>' + fmt('Package', 'Date       Time', OLD_VERSION, 'Head', None) + '</b>\n'
+    html += '<b>' + fmt('Package', 'Date       Time', OLD_VERSION, 'Head', link=False) + '</b>\n'
     current_year = datetime.date.today().year
+    stack_traces = {}
     for filename in sorted(glob.glob(os.path.expanduser(results_path + '/*'))):
         if not os.path.isfile(filename):
             continue
         datestr = ''
-        for line in open(filename, 'rt'):
-            line = line.strip()
-            if line.startswith('cppcheck: '):
-                if OLD_VERSION not in line:
-                    # Package results seem to be too old, skip
+        with open(filename, 'rt') as file:
+            for line in file:
+                line = line.strip()
+                if line.startswith('cppcheck: '):
+                    if OLD_VERSION not in line:
+                        # Package results seem to be too old, skip
+                        break
+                    else:
+                        # Current package, parse on
+                        continue
+                if line.startswith(str(current_year) + '-') or line.startswith(str(current_year - 1) + '-'):
+                    datestr = line
+                if line.startswith('count:'):
+                    if line.find('Crash') < 0:
+                        break
+                    package = filename[filename.rfind('/')+1:]
+                    counts = line.strip().split(' ')
+                    c2 = ''
+                    if counts[2] == 'Crash!':
+                        c2 = 'Crash'
+                    c1 = ''
+                    if counts[1] == 'Crash!':
+                        c1 = 'Crash'
+                    html += fmt(package, datestr, c2, c1) + '\n'
+                    if c1 != 'Crash':
+                        break
+                if line.find(' received signal ') != -1:
+                    crash_line = next(file, '').strip()
+                    location_index = crash_line.rindex(' at ')
+                    if location_index > 0:
+                        code_line = next(file, '').strip();
+                        stack_trace = []
+                        while True:
+                            l = next(file, '')
+                            m = re.search('(?P<number>#\d+) .* (?P<function>.+)\(.*\) at (?P<location>.*)$', l)
```

Comment:
I get these PEP8 messages for this regular expression:
PEP 8: invalid escape sequence '\('
PEP 8: invalid escape sequence '\)'
PEP 8: invalid escape sequence '\d'

Reply:
According to https://youtrack.jetbrains.com/issue/PY-37740 and https://lintlyci.github.io/Flake8Rules/rules/W605.html a raw string solves this issue.
The [python documentation ](https://docs.python.org/2.7/library/re.html) also suggests using raw strings when using escape sequences.
```suggestion
                            m = re.search(r'(?P<number>#\d+) .* (?P<function>.+)\(.*\) at (?P<location>.*)$', l)
```